src/util/address.rs

```diff
@@ -205,8 +205,125 @@ impl Payload {
             } => script::Script::new_witness_program(ver, prog)
         }
     }
+
+    /// Creates a pay to (compressed) public key hash address payload from a public key.
+    pub fn p2pkh(pk: &ecdsa::PublicKey) -> Payload {
+        let mut hash_engine = PubkeyHash::engine();
+        pk.write_into(&mut hash_engine)
+            .expect("engines don't error");
+        Payload::PubkeyHash(PubkeyHash::from_engine(hash_engine))
+    }
+
+    /// Creates a pay to script hash P2SH address payload from a script
+    #[inline]
+    pub fn p2sh(script: &script::Script) -> Payload {
```

Comment:
We should also error if the script is more than (MAX_SCRIPT_ELEMENT_SIZE) 520 bytes.

Reply:
I just note that this is unrelated to the PR and can be ignored.

Reply:
Created an issue for that https://github.com/rust-bitcoin/rust-bitcoin/issues/650